streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java

```diff
@@ -821,4 +823,20 @@ public Options getOptions() {
     public Position getPosition() {
         return position;
     }
+
+    /**
+     * Same as {@link Bytes#increment(Bytes)} but {@code null} is returned instead of throwing
+     * {@code IndexOutOfBoundsException} in the event of overflow.
+     *
+     * @param input bytes to increment
+     * @return A new copy of the incremented byte array, or {@code null} if incrementing would
+     *         result in overflow.
+     */
+    static Bytes incrementWithoutOverflow(final Bytes input) {
```

Comment:
Now that negative segment ID is allowed (only for reserved segments within `LogicalKeyValueSegments`), it's possible that a valid segment ID will overflow when incremented, and we want to handle this gracefully instead of throwing an exception when it happens.